SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp

```diff
@@ -166,112 +169,118 @@ void ComponentReader::ApplySchemaObject(Schema_Object* ComponentObject, UObject&
+				uint8* Data = (uint8*)&Object + SwappedCmd.Offset;
 
-					if (ValueData.Num() > 0)
+				if (Cmd.Type == ERepLayoutCmdType::DynamicArray)
+				{
+					UArrayProperty* ArrayProperty = Cast<UArrayProperty>(Cmd.Property);
+					if (ArrayProperty == nullptr)
 					{
-						FSpatialNetDeltaSerializeInfo::DeltaSerializeRead(NetDriver, ValueDataReader, &Object, Parent.ArrayIndex, Parent.Property, NetDeltaStruct);
+						UE_LOG(LogSpatialComponentReader, Error, TEXT("Failed to apply Schema Object %s. One of it's properties is null"), *Object.GetName());
+						continue;
 					}
 
-					FObjectReferences* CurEntry = RootObjectReferencesMap.Find(SwappedCmd.Offset);
-					const bool bHasReferences = NewUnresolvedRefs.Num() > 0 || NewMappedRefs.Num() > 0;
-
-					if (ReferencesChanged(RootObjectReferencesMap, SwappedCmd.Offset, bHasReferences, NewMappedRefs, NewUnresolvedRefs))
+					// Check if this is a FastArraySerializer array and if so, call our custom delta serialization
+					if (UScriptStruct* NetDeltaStruct = GetFastArraySerializerProperty(ArrayProperty))
 					{
-						if (bHasReferences)
+						SCOPE_CYCLE_COUNTER(STAT_ReaderApplyFastArrayUpdate);
+
+						TArray<uint8> ValueData = GetBytesFromSchema(ComponentObject, FieldId);
+						int64 CountBits = ValueData.Num() * 8;
+						TSet<FUnrealObjectRef> NewMappedRefs;
+						TSet<FUnrealObjectRef> NewUnresolvedRefs;
+						FSpatialNetBitReader ValueDataReader(PackageMap, ValueData.GetData(), CountBits, NewMappedRefs, NewUnresolvedRefs);
+
+						if (ValueData.Num() > 0)
 						{
-							RootObjectReferencesMap.Add(SwappedCmd.Offset, FObjectReferences(ValueData, CountBits, MoveTemp(NewMappedRefs), MoveTemp(NewUnresolvedRefs), ShadowOffset, Cmd.ParentIndex, ArrayProperty, /* bFastArrayProp */ true));
+							FSpatialNetDeltaSerializeInfo::DeltaSerializeRead(NetDriver, ValueDataReader, &Object, Parent.ArrayIndex, Parent.Property, NetDeltaStruct);
 						}
-						else
+
+						FObjectReferences* CurEntry = RootObjectReferencesMap.Find(SwappedCmd.Offset);
+						const bool bHasReferences = NewUnresolvedRefs.Num() > 0 || NewMappedRefs.Num() > 0;
+
+						if (ReferencesChanged(RootObjectReferencesMap, SwappedCmd.Offset, bHasReferences, NewMappedRefs, NewUnresolvedRefs))
 						{
-							RootObjectReferencesMap.Remove(SwappedCmd.Offset);
+							if (bHasReferences)
+							{
+								RootObjectReferencesMap.Add(SwappedCmd.Offset, FObjectReferences(ValueData, CountBits, MoveTemp(NewMappedRefs), MoveTemp(NewUnresolvedRefs), ShadowOffset, Cmd.ParentIndex, ArrayProperty, /* bFastArrayProp */ true));
+							}
+							else
+							{
+								RootObjectReferencesMap.Remove(SwappedCmd.Offset);
+							}
+							bOutReferencesChanged = true;
 						}
-						bOutReferencesChanged = true;
+					}
+					else
+					{
+						ApplyArray(ComponentObject, FieldId, RootObjectReferencesMap, ArrayProperty, Data, SwappedCmd.Offset, ShadowOffset, Cmd.ParentIndex, bOutReferencesChanged);
```

Comment:
Do we also want to track this and ApplyProperty below?

Reply:
Sure